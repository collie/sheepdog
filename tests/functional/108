#!/bin/bash

# Test sheep -p/--port

. ./common

err=0

function testStartPortSucceeded
{
	port="$1"
	mkdir -p "$STORE/0"
	if $SHEEP "$STORE/0" -p "$port" ; then
		local i=0
		while [ $i -lt 5 ] ; do
			if grep daemon "$STORE/0/sheep.log" >/dev/null 2>&1 ; then
				$DOG node list -p "$port"
				break
			fi
			sleep 1
			i=$((i+1))
		done
		if [ $i -eq 5 ] ; then
			err=1
		fi
	else
		err=1
	fi
	_cleanup
}

function testStartPortFailed
{
	port="$1"
	mkdir -p "$STORE/0"
	$SHEEP "$STORE/0" -p "$port" && err=1
	_cleanup
}

testStartPortSucceeded 7000
testStartPortFailed 0
testStartPortFailed 65536 # UINT16_MAX + 1
testStartPortFailed -1
testStartPortFailed a
testStartPortFailed 42a

exit $err
