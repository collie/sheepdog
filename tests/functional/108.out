QA output created by 108
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128   16777343
Invalid port number '0'
Invalid port number '65536'
Invalid port number '-1'
Invalid port number 'a'
Invalid port number '42a'
